From 579cf314cbdf37d7934d2787a210beee8d1fbc82 Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Thu, 11 Jun 2015 11:05:42 +0800
Subject: [PATCH 422/462] ov8858: add v1p2 core power control

  + There are 3 power lines for rear sensor. V1P2 provides
core power supply on FFD.

  + Fixed an unexpected err print

Change-Id: I0a99f2393d7762aa78f330f4563ba838911ac632
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-37622
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/59896
Build: IMIN (SC)
Approver: Jian Xu Zheng <jian.xu.zheng@intel.com>
Reviewed-by: Jian Xu Zheng <jian.xu.zheng@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index c3968827..9fcf171 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -676,7 +676,7 @@ static long ov8858_ioctl(struct v4l2_subdev *sd, unsigned int cmd, void *arg)
 	case ATOMISP_IOC_G_DEPTH_SYNC_COMP:
 		return ov8858_g_comp_delay(sd, (unsigned int *)arg);
 	default:
-		dev_err(&client->dev, "Unhandled command 0x%X\n", cmd);
+		dev_dbg(&client->dev, "Unhandled command 0x%X\n", cmd);
 		return -EINVAL;
 	}
 }
@@ -697,6 +697,16 @@ static int __power_ctrl(struct v4l2_subdev *sd, bool flag)
 		return dev->platform_data->power_ctrl(sd, flag);
 
 #ifdef CONFIG_GMIN_INTEL_MID
+	if (dev->platform_data->v1p2_ctrl) {
+		ret = dev->platform_data->v1p2_ctrl(sd, flag);
+		if (ret) {
+			dev_err(&client->dev,
+				"failed to power %s 1.2v power rail\n",
+				flag ? "up" : "down");
+			return ret;
+		}
+	}
+
 	if (dev->platform_data->v2p8_ctrl) {
 		ret = dev->platform_data->v2p8_ctrl(sd, flag);
 		if (ret) {
-- 
1.9.1

